### PR TITLE
Exclude itself from the connection list [ECR-4190]

### PR DESCRIPTION
**Pull request by aleksuss**
## Overview

---

See: https://jira.bf.local/browse/ECR-4190

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: cli/src/command/finalize.rs [hunk 1]
```diff
@@ -170,6 +184,7 @@ impl ExonumCommand for Finalize {
                 ..private_config.api
             },
             connect_list,
+            consensus_key: None, // We use it in creation connect list. We don't need it anymore.
```
Comment (vitvakatu) [inline cli/src/command/finalize.rs]: Is there a good reason _not_ to store it in the config (i.e. get rid of `Option`)?

File: cli/src/command/finalize.rs [hunk 2]
```diff
@@ -114,13 +114,24 @@ impl Finalize {
             .consensus_key)
     }
 
-    fn create_connect_list_config(public_configs: &[NodePublicConfig]) -> ConnectListConfig {
-        // FIXME: filter self (ECR-4190).
+    fn create_connect_list_config(
+        public_configs: &[NodePublicConfig],
+        skipped_key: &PublicKey,
```
Comment (vitvakatu) [inline cli/src/command/finalize.rs]: nit: `skip_key` is more a bit more logical variable name IMO

File: cli/src/command/finalize.rs (hunk 1 above)
Comment (vitvakatu) [reply cli/src/command/finalize.rs]: Also you can consider renaming it into `consensus_public_key`, as it is not clear when reading configs.

Comment (aleksuss) [reply cli/src/command/finalize.rs]: The reason not to store is because of data duplication. We already have consensus key in *validator_keys* array.

Comment (vitvakatu) [reply cli/src/command/finalize.rs]: Is it bad? Right now we do not have a way to distinguish which private config corresponds to a public config.

File: cli/src/command/finalize.rs (hunk 2 above)
Comment (slowli) [reply cli/src/command/finalize.rs]: @vitvakatu `skip_key` is a verbal phrase though; IMO, noun phrases are more appropriate for variable names.

File: cli/src/command/finalize.rs (hunk 1 above)
Comment (slowli) [reply cli/src/command/finalize.rs]: I agree with @vitvakatu that `consensus_public_key` would be a better name. Regarding duplication, I don't see it as a significant problem; for example, a node may be an auditor, in which case `consensus_public_key` is not duplicated. (As far as I understand, `exonum-cli` does not have first-class auditor support during blockchain creation, but this is a subject of future improvements.)

File: cli/src/command/finalize.rs (hunk 2 above)
Comment (vitvakatu) [reply cli/src/command/finalize.rs]: `skipped_key` means "the key that was skipped", while the actual meaning is "the key that is needed to be skipped". As `key_to_skip` is a bit long, I suggested to use `skip_key`, which is used nicely in a function call like `create_connect_list_config(..., skip_key: key)`. However, I'm not gonna to argue about variables names, it was a nit.

Comment (popzxc) [reply cli/src/command/finalize.rs]: IMHO `key_to_skip` is just fine.

**Pull request merged**